packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-questions.ts

```diff
@@ -92,7 +92,7 @@ export async function askBucketNameQuestion(context: $TSContext, defaultValues:
   const message = 'Provide bucket name:';
   const onErrorMsg =
     'Bucket name can only use the following characters: a-z 0-9 - and should have minimum 3 character and max of 47 character';
-  const validator = and([alphanumeric(), minLength(3), maxLength(47)], onErrorMsg);
+  const validator = and([regexValidator('Input can only contain the following characters: a-z 0-9', RegExp(/^[a-z0-9-]+$/) ),minLength(3), maxLength(47)], onErrorMsg);
```

Comment:
Can we double check the regex? I believe hyphen(-) has a special meaning so it may be required to escape like this `RegExp(/^[a-z0-9\-]+$/)`.

Reply:
Also nit: don't need to use the RegExp constructor when using slashes. I would recommend just using slashes

Reply:
@sundersc I have tested this both in the unit test ( s3 and validator) and manual (bucket creation). we used this regex in the inquirer library, hence its good.